logger/include/logger.hpp

```diff
@@ -1,20 +1,49 @@
 #pragma once
 
+#include <filesystem>
 #include <iostream>
+#include <list>
+#include <map>
 #include <source_location>
+#include <sstream>
 
-namespace logging
+namespace Logger
 {
 
+enum Loglevel
+{
+    INFO,
+    CRITICAL
+};
+
+// Maximum size of buffer is 8k bytes
+const size_t maxBufferSize = 8000;
+inline std::list<std::string> circularBuffer;
+
+/**
+ *@brief Appends data to the buffer
+ *
+ *This API is called to log message to the buffer.
+ *It checks the log level and appends information like
+ *line number, file name, timestamp to the message.
+ *
+ *@param[in] level - loglevel of the message
+ *@param[in] message - Information to be logged
+ *@param[in] location - object of the source_location class
+ */
+void logMessage(
+    const Loglevel level, const std::string& message,
+    const std::source_location& location = std::source_location::current());
+
 /**
- *@brief An API to append log messages to a buffer
- *This API is called to log message. It also appends the
- *file name and line number of the mesaage.
+ *@brief calls the logMessage(level, message, location) with INFO as default
```

Comment:
Please wrap brief in one line.
The API is to log message. What it calls internally need not be defined in brief. Though you can add some details after brief

Eg: An API to log message with log level defaulted to info.

Reply:
done